src/suggestion.rs

```diff
@@ -50,6 +92,151 @@ impl fmt::Display for Detector {
+    //
+    //|---------------------------------------------------------------------------------------| terminal_size
+    //|-------| padding_till_literal_start = indent (3+line_number_digit_count) + 2 white spaces = 7usize, for this case.
+    //     |------| offset = PADDING_OFFSET; 3 chars for `...` and 2 white spaces more added in the formatting.
+    //
+    //   --> /home/tmhdev/Documents/cargo-spellcheck/src/suggestion.rs:62
+    //    |
+    // 62 |  ... Reasn of food, what's up with pie? There's strawberry pie, apple, pumpkin ...
+    //    |      ^^^^^^
+    //    | - there, Cherie, thither, or tither
+    //    |
+    //    |   Possible spelling mistake found.
+    //
+
+    const PADDING_OFFSET: usize = 5;
+    const TOO_LONG_WORD: usize = 20;
+    const DISPLAYED_LONG_WORD: usize = 4;
+    const PADDING_AROUND_LONG_LINES: usize = 10;
+
+    // We will be using ranges to help doing the fitting:
+    //
+    // |----------------------------------literal_word----------------------------------|
+    // |----------------------|---------misspelled_word---------|-----------------------|
+    // |-----left_context-----|---start_word----|----end_word---|-----right_context-----|
+    //
+    // Obs: paddings are not being considered in the illustration, but info is above.
+
+    // the line being analysed can affect how the indentation is done
+    // this values is dynamically calculated according to the line number
+    let mut range_start_word = Range {
+        start: range_word.start,
+        end: range_word.start,
+    };
+    let mut range_end_word = Range {
+        start: range_word.end,
+        end: range_word.end,
+    };
+    let mut misspelled_word: String = stripped_line
+        .chars()
+        .skip(range_word.start)
+        .take(range_word.len())
+        .collect();
+    // Check words that are considered too long; Word will be formatted for fitting
+    if range_word.len() > TOO_LONG_WORD {
+        range_start_word = Range {
+            start: range_word.start,
+            end: range_start_word.start + DISPLAYED_LONG_WORD,
+        };
+        range_end_word = Range {
+            start: range_word
+                .end //non inclusive
+                .saturating_sub(DISPLAYED_LONG_WORD),
+            end: range_word.end,
+        };
+
+        //  too long word will be shorter as it follows:
+        //    4 chars |----|  ... |---| 3 chars
+        //                ther...eee, for therieeeeeeeeeeeeeeee
+        //
+        misspelled_word = format!(
+            "{}...{}",
+            stripped_line
+                .chars()
+                .skip(range_start_word.start)
+                .take(range_start_word.len())
+                .collect::<String>(),
+            stripped_line
+                .chars()
+                .skip(range_start_word.end.saturating_sub(3))
+                .take(3)
+                .collect::<String>()
+        );
+        *marker_size = misspelled_word.chars().count();
```

Comment:
use them here to avoid another `O(n)` operation

Reply:
I still need to count their sizes again here per chars, right?